### PR TITLE
Update the wasm-tools family of crates, disallow empty component types

### PR DESCRIPTION
**Pull request by alexcrichton**
This commit updates the wasm-tools family of crates which notably brings in a change to disallow empty types in the component model. There were a number of tests and fuzz-related things to update, but the "main" change here is that the `stream-error` type in WASI is growing a `dummy: u32` field as a simple way to keep things compiling. This'll also hopefully emphasize that this is a transitionary state where it'll get updated in the future to be a non-empty type.

### CODE REVIEW

File: crates/wasi/wit/deps/io/streams.wit [hunk 1]
```diff
@@ -8,9 +8,15 @@ package wasi:io
 interface streams {
     use wasi:poll/poll.{pollable}
 
-    /// An error type returned from a stream operation. Currently this
-    /// doesn't provide any additional information.
-    record stream-error {}
+    /// An error type returned from a stream operation.
+    ///
+    /// TODO: need to figure out the actual contents of this eror. Used to be
```
Comment (peterhuene) [inline crates/wasi/wit/deps/io/streams.wit:13]:
```suggestion
    /// TODO: need to figure out the actual contents of this error. Used to be
```

**Review by pchickey — APPROVED** [review]
I had never noticed that dead wit code, thanks

File: crates/test-programs/reactor-tests/wit/deps/filesystem/filesystem.wit
Comment (peterhuene) [inline crates/test-programs/reactor-tests/wit/deps/filesystem/filesystem.wit:1]: For other reviewers: these wit files are no longer needed as the test programs already put it from the `crates/wasi/wit` directory.

**Review by peterhuene — APPROVED** [review]

**Comment by peterhuene** [comment]
Looks like Pat beat me to the review after all 😅 ; I was going to sneak in a review quickly so we can get this merged as the WASI adapters no longer validate with the latest wasm-tools, preventing me from bumping the deps for `cargo-component`.

**Pull request merged**